src/cart/crt_register.c

```diff
@@ -629,7 +629,7 @@ crt_proto_query_int(crt_endpoint_t *tgt_ep, crt_opcode_t base_opc, uint32_t *ver
 	proto_query->pq_user_arg = arg;
 	proto_query->pq_coq->coq_base = base_opc;
 
-	rc = crt_req_set_timeout(rpc_req, 5);
+	rc = crt_req_set_timeout(rpc_req, timeout);
```

Comment:
I would prefer here if timeout passed is 0 to not crt_req_set_timeout() at all. This way callers like crt_rpc.c can then call with 0 and use whatever defaults were already set via CRT_TIMEOUT instead of having to specify 60 seconds (for fi/ctl proto queries below).

Reply:
you mean timeout passed from the cart tests, but not the daos proto query from init? i agree in that case that passing 0 should just be a no-op and not do anything in this case.